### ISSUE TITLE
Examples crash on Metal

### ISSUE THREAD

**Issue by Gordon-F** [hecrj/coffee#125]
Example crash on Metal.

Tested examples:
1. UI
2. Rectangle
3. Snake

version: master branch (c42d269776dc3af1b1dc8dd9aee213ad9ee9e674)

```
RUST_BACKTRACE=1 METAL_DEVICE_WRAPPER_TYPE=1 RUST_LOG=info cargo run --example ui --features metal
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/examples/ui`
2020-05-09 12:38:04.986 ui[9708:178662] Metal API Validation Enabled
thread 'main' panicked at 'assertion failed: image::Usage::COLOR_ATTACHMENT.contains(config.image_usage)', /Users/indish/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-metal-0.5.1/src/window.rs:463:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1063
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:470
  11: rayon_core::sleep::Sleep::tickle_cold
  12: <gfx_backend_metal::window::Surface as gfx_hal::window::PresentationSurface<gfx_backend_metal::Backend>>::configure_swapchain
             at /Users/indish/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-metal-0.5.1/src/window.rs:463
  13: wgpu_core::device::<impl wgpu_core::hub::Global<G>>::device_create_swap_chain
             at /Users/indish/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.5.4/src/device/mod.rs:2026
  14: wgpu_device_create_swap_chain
             at /Users/indish/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-native-0.5.0/src/device.rs:356
  15: wgpu::Device::create_swap_chain
             at /Users/indish/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.5.0/src/lib.rs:1005
  16: coffee::graphics::backend_wgpu::surface::new_swap_chain
             at src/graphics/backend_wgpu/surface.rs:86
  17: coffee::graphics::backend_wgpu::surface::Surface::new
             at src/graphics/backend_wgpu/surface.rs:19
  18: coffee::graphics::backend_wgpu::Gpu::for_window
             at src/graphics/backend_wgpu/mod.rs:60
  19: coffee::graphics::window::Window::new
             at src/graphics/window.rs:37
  20: coffee::game::loop::Loop::run
             at ./src/game/loop.rs:41
  21: coffee::ui::UserInterface::run
             at ./src/ui.rs:274
  22: ui::main
             at examples/ui.rs:12
  23: std::rt::lang_start::{{closure}}
             at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/rt.rs:67
  24: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  25: std::panicking::try::do_call
             at src/libstd/panicking.rs:303
  26: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:86
  27: std::panicking::try
             at src/libstd/panicking.rs:281
  28: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  29: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  30: std::rt::lang_start
             at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/rt.rs:67
  31: <ui::Layout as core::fmt::Debug>::fmt
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

All works fine if i remove COPY_DST from swapchain usage.
```diff
diff --git a/src/graphics/backend_wgpu/surface.rs b/src/graphics/backend_wgpu/surface.rs
index d823781..c2363b5 100644
--- a/src/graphics/backend_wgpu/surface.rs
+++ b/src/graphics/backend_wgpu/surface.rs
@@ -86,8 +86,7 @@ fn new_swap_chain(
     let swap_chain = device.create_swap_chain(
         surface,
         &wgpu::SwapChainDescriptor {
-            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT
-                | wgpu::TextureUsage::COPY_DST,
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
             format: wgpu::TextureFormat::Bgra8UnormSrgb,
             width: size.width,
             height: size.height,
```

**Comment by hecrj** [comment]
Good catch! `COPY_DST` is no longer necessary.

Would you mind opening a PR with the fix?

**Comment by Gordon-F** [comment]
@hecrj fix this in #126
